build-system/server/typescript-compile.js

```diff
@@ -14,29 +14,38 @@
  * limitations under the License.
  */
 const log = require('fancy-log');
+const pathModule = require('path');
 const {cyan, green} = require('ansi-colors');
 const {exec} = require('../common/exec');
 
 const SERVER_TRANSFORM_PATH = 'build-system/server/new-server/transforms';
 
-// Used by new server implementation
-const typescriptBinary = './node_modules/typescript/bin/tsc';
+function getBuildCmd() {
+  switch (process.platform) {
+    case 'win32':
+      return `node .\\node_modules\\typescript\\lib\\tsc.js -p ${SERVER_TRANSFORM_PATH.split(
+        '/'
+      ).join(pathModule.sep)}${pathModule.sep}tsconfig.json`;
```

Comment:
I think it might be possible to replace this split / join block with a [`path.posix`](https://nodejs.org/api/path.html#path_path_posix) method.

/cc @samouri, who has experience with this, and the vagaries of Windows.